src/include/duckdb/main/client_context_wrapper.hpp

```diff
@@ -18,6 +18,9 @@ class ClientContextWrapper {
 public:
 	explicit ClientContextWrapper(const shared_ptr<ClientContext> &context);
 	shared_ptr<ClientContext> GetContext();
+	operator bool() const {
+		return !client_context.expired();
```

Comment:
Is this required? Generally checking `expired` on a weak_ptr is an anti-pattern, since there is no guarantee that the shared pointer is not destroyed before executing any code aftwards

Reply:
Yea it not being atomic is not great

The problem though is that we hide the weak_ptr here, and the only way to access it otherwise is through GetContext, which throws if the context is gone.

If I'm doing a replacement scan on a relation, that error isn't very descriptive in context

Reply:
Maybe add a `TryGetContext`?

Reply:
It doesn't matter in this case, but it's just bad practice to use `expired()` since doing that is essentially always a race condition.